src/server/handlers/wallet/balances.rs

```diff
@@ -46,7 +48,11 @@ impl JsonRPCRequestHandler for WalletListHandler {
         };
 
         let manager = conn.manager();
-        let addresses = manager.wallet_list().await?;
+
+        let addresses = self.wallet.read().unwrap().list_addrs()?;
+        // Create a new Arc for wallet so it is pulled in the async block
+        // from below.
+        let _arc_wallet = Arc::clone(&self.wallet);
```

Comment:
do we still need this if `_arc_wallet` is not used at all?

Reply:
Unfortunately, we do because we use an `async_move` in the next line and if we don´t do this the wallet won't implement `Send` and the compiler won't be happy. This makes it happy :)